### ISSUE TITLE
Consider adding an equivalence of EXPOSE

### ISSUE THREAD

**Issue by tcnghia** [google/ko#472]
Per https://docs.docker.com/engine/reference/builder/#expose:

> The EXPOSE instruction does not actually publish the port. It functions as a type of documentation between the person who builds the image and the person who runs the container, about which ports are intended to be published. 

Some container service (like Azure App Service) consumes this metadata to automatically configure the correct port-mapping to the container. Without this metadata, users of these service will need to set this on the deployment side.

cc @jonjohnsonjr

**Comment by jonjohnsonjr** [comment]
Yep yep, 100% this is the goal. Given that goreleaser does an excellent job of exposing everything you might need to configure for a go build, we just stole it rather than reinventing the wheel. As long as goreleaser is able to ignore unknown fields (e.g. the ko config), you should be able to use the same config file for both tools.

One kind of strange bit is that `ko` doesn't use goreleaser to drive the builds themselves -- we have implemented a bunch of this logic on our own. (Thinking out loud) I wonder if it would be possible to pull out part of goreleaser into a package such that `ko` could just depend on it directly and dispatch to goreleaser for the builds and only handle the container packaging bits.

I think [this](https://goreleaser.com/customization/docker/) makes things a little weird, and also some ko-isms like bundling `kodata/`, but maybe that's fine if we can slice up the dependencies appropriately. I'm just trying to avoid circular dependencies without making this too complicated.

(Probably time to pull this into a separate issue...)

**Comment by mbrancato** [comment]
Coming here to comment that when using tools like Knative without the `EXPOSE` functionality, we cannot hit things like pod metric endpoint directly. This means we get the proxy sidecar as the "container" label of what is reporting the metric, so it looks incorrect.
